subsys/bluetooth/host/l2cap.c

```diff
@@ -2980,10 +2992,13 @@ int bt_l2cap_ecred_chan_reconfigure(struct bt_l2cap_chan **chans, uint16_t mtu)
 		if (mtu < BT_L2CAP_LE_CHAN(chans[i])->rx.mtu) {
 			return -EINVAL;
 		}
-	}
 
-	if (i == 0) {
-		return -EINVAL;
+		/* MPS is not allowed to decrease when reconfiguring multiple channels.
+		 * Core Specification 3.A.4.27 v6.0
+		 */
+		if (multiple_chan && mps < BT_L2CAP_LE_CHAN(chans[i])->rx.mps) {
+			return -EINVAL;
```

Comment:
Alternative solution without the added `bool` variable

```suggestion
		if (i > 0 && mps < BT_L2CAP_LE_CHAN(chans[i])->rx.mps) {
			return -EINVAL;
```